kolibri/content/api.py

```diff
@@ -222,6 +223,20 @@ def filter_kind(self, queryset, name, value):
             return queryset.exclude(kind=content_kinds.TOPIC).order_by("lft")
         return queryset.filter(kind=value).order_by("lft")
 
+    def filter_by_role(self, queryset, name, value):
+        """
+        Show coach_content if they have coach role or higher.
+
+        :param queryset: all content nodes for this channel
+        :param value: 'boolean'
+        :return: content nodes filtered by coach_content if appropiate
+        """
+        user = self.request.user
+        if user.is_facility_user:  # exclude anon users
+            if user.roles.exists() or user.is_superuser:  # must have coach role or higher
+                return queryset
+        return queryset.exclude(coach_content=True)
```

Comment:
@jayoshih This might lead to unexpected behavior on hand-annotated Channels like firki. My understanding is that Topics should not have `coach_content == True`, which is the case for Firki. So this user would not be able to see this topic, even if it did have non coach-content stuff in it.

Reply:
@ralphiee22 I think we'll see how this works out. If we are consistent with the no-topic-should-be-coach-content policy, I think this will work in general. Otherwise, we might need to refine this queryset.

Reply:
@jonboiser Right now, it is impossible to set the role for topics (Firki's topics all are set to the learner visibility, even though all of the non-topics have the coach flag). This is something we need to explore in the future for sure though- not sure when we plan to do a second pass though

Reply:
I think on studio when saving the nodes we just always set `coach_content=False` for `topic` nodes, or don't show that option in the studio UI. @jayoshih

Reply:
Or, alternatively, if we do allow it to be set on topics, we should either warn the curator that this will mean any content in the topic will be invisible to users, or automatically propagate this to everything under the topic to remove the ambiguity.

Reply:
hmm, so I was wrong and topics in Firki do have coach_content = True. I had thought otherwise because I was constructing a queryset to get all descendants of a Firki topic with Coach_contents = true and was getting Topic nodes in the result. Maybe that's something unrelated?